cmd/promtool/main.go

```diff
@@ -14,18 +14,123 @@
 package main
 
 import (
+	"flag"
 	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strconv"
 	"strings"
+	"time"
 
+	"golang.org/x/net/context"
+
+	"github.com/prometheus/client_golang/api/prometheus"
 	"github.com/prometheus/common/version"
 	"github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/promql"
 	"github.com/prometheus/prometheus/util/cli"
 )
 
+var (
+	server  = flag.String("server", "http://localhost:9090", "URL of the Prometheus server to query (default http://localhost:9090)")
+	timeout = flag.Duration("timeout", time.Minute, "Timeout to use when querying the Prometheus server (default 1m0s)")
+)
+
+// Query returns metrics from the prometheus server.
+func Query(t cli.Term, args ...string) int {
+	if len(args) == 0 {
+		t.Infof("usage: promtool [flags] query <expression>")
+		return 2
+	}
+
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, *timeout)
+	defer cancel()
+
+	client, err := prometheus.New(prometheus.Config{
+		Address: *server,
+	})
+	if err != nil {
+		t.Errorf(" FAILED: %s", err)
+		return 1
+	}
+
+	api := prometheus.NewQueryAPI(client)
+	result, err := api.Query(ctx, args[0], time.Now())
+	if err != nil {
+		t.Errorf(" FAILED: %s", err)
+		return 1
+	}
+
+	fmt.Println(result)
+	return 0
+}
+
+// QueryRange returns range metrics from the prometheus server.
+func QueryRange(t cli.Term, args ...string) int {
+	if len(args) < 3 {
+		t.Infof("usage: promtool [flags] query_range <expression> <end_timestamp> <range_seconds> [<step_seconds>]")
```

Comment:
It'd be a lot more convenient to provide default values for end_timestamp and range_seconds. I'd say everything but the expression should be flags and not positional arguments.